appserver/extras/payara-micro/payara-micro-boot/src/main/java/fish/payara/micro/boot/loader/ExplodedURLClassloader.java

```diff
@@ -73,14 +75,13 @@ public ExplodedURLClassloader(File explodeTo) throws IOException {
 
     public ExplodedURLClassloader() throws IOException {
         super(new URL[0]);
-        File directory;
-        directory = File.createTempFile("payaramicro-rt", "tmp");
+        File directory = File.createTempFile("payaramicro-rt", "tmp");
```

Comment:
This is why we need recursive deletion of files and directories ... ok

Reply:
But ... the directory variable will be a regular file, not a directory. See Files.createTempDirectory

Reply:
> But ... the directory variable will be a regular file, not a directory. See Files.createTempDirectory

Note, that only generated name is used, as in next line https://github.com/payara/Payara/blob/5b1ff95c8e01be695be8cddb482b43270dc37957/appserver/extras/payara-micro/payara-micro-boot/src/main/java/fish/payara/micro/boot/loader/ExplodedURLClassloader.java#L79 there is attemp to _convert_ it to directory.

Reply:
You're right ... I forgot. Just another argument to use NIO instead of old io... it simplifies many things.

Reply:
Sure, I can switch it to NIO. Recursion is a bit tricky to read.